rtt_ros2_topics/include/orocos/rtt_ros2_topics/topic.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Intermodalics BVBA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OROCOS__RTT_ROS2_TOPICS__TOPIC_HPP_
+#define OROCOS__RTT_ROS2_TOPICS__TOPIC_HPP_
+
+#include <string>
+#include <utility>
+
+#include "rtt/ConnPolicy.hpp"
+
+#include "rtt_ros2_topics/protocol_id.h"
+
+namespace rtt_ros2_topics
+{
+
+/// Returns a ConnPolicy object for streaming to or from the given ROS topic.
+/**
+ * Only a single data object is buffered within RTT and ROS, which replaces all earlier samples
+ * that have not been actually published or read yet.
+ *
+ * @param topic The topic (copied to field name_id).
+ *              If empty, use the port name converted to lower case with underscores.
+ * @param latch If true, new subscribers will see old messages published before, i.e.
+ *              set durability to transient local (copied to field init).
+ */
+static RTT::ConnPolicy topic(std::string topic = std::string(), bool latch = false)
+{
+  RTT::ConnPolicy policy = RTT::ConnPolicy::data();
+  policy.init = latch;
+  policy.name_id = std::move(topic);
```

Comment:
If you don't make a copy, pass complex types by const-reference.
If you want to retain a copy (like in this case), pass by value and `std::move`.
Passing by lvalue-reference as you suggested is only valid if you want to modify `topic` in the caller's namespace.

Furthermore only accepting non-const lvalues as arguments would limit the possible scopes of the function. For example, this example would not compile:
```cpp
auto policy = rtt_ros2_topics::topic("chatter");
```
because `"chatter"` is an rvalue and cannot be stored in an lvalue-reference.

What would be possible is
```cpp
static RTT::ConnPolicy topic(const std::string & topic = std::string(), bool latch = false)
{
  RTT::ConnPolicy policy = RTT::ConnPolicy::data();
  policy.init = latch;
  policy.name_id = topic;
  policy.transport = ORO_ROS2_PROTOCOL_ID;
  return policy;
}
```
but then for the first example the `topic` string is copied twice (unless [small string](https://shaharmike.com/cpp/std-string/) or other optimizations kick in): Once when constructing the temporary `std::string` in the caller and once more to assign to `policy.name_id`. In this case it is much more likely that the function gets called with a temporary string, in which case the pass-by-value is preferable.